README.md
# Examen 372 - Terminal / Git-Github 👻💻

 > " Solo hay un riesgo que deberías evitar a toda costa, y es el riesgo de no hacer nada 💪🔥" *(Denis Waitley)*


### Instrucciones Generales 🤔: 

1. Todas las instrucciones deben ser ejecutadas en terminal a menos que se indique lo contrario.
2. Se permite el uso de internet alias el **"googlazo"**.
3. La entrega del examen es un repositorio en su cuenta de GITHUB con la solución.
4. Cada inciso del examen debe ser marcado como completado para verificar que se ha realizado correctamente.
5. Enviar correo a iavila24@uabc.edu.mx con el link del repositorio para verificar examen.
6. Límite de entrega **`24-OCT-2023 20:59:59`**
---

### Intrucciones Examen 🧐:

- [x] Coloca tu nombre aqui: *`"Jesus Hernandez Ruiz"`*
- [x] Abre el shell/terminal *`"Git Bash"`*
- [x] Limpia el historial de tu consola: *`"history -c"`*
- [x] Navega al directorio home del usuario. *`"~"`* 
- [x] Crear una carpeta con el nombre. *`"examen372"`* y colocate dentro de ella.
- [x] Crear un script con el nombre. *`"create.sh"`*
- [x] Colocar el shebang dentro del script desde la terminal: *`"#! /bin/bash"`*
- [x] Inicializa un repositorio vacio de git.
- [x] Crea un commit inicial con el mensaje *`"Initial commit, script added"`* (el commit debe contener la version inicial del script).
---

### Instrucciones dentro del script `create.sh` 🔡:

1. [x] Elimina la carpeta **webapp**. *`"rm -rf"`*
2. [x] Crea la siguiente estructura de carpetas *`"mkdir"`*:
  
    ```
    .
    └── webapp/
        ├── css
        ├── js
        ├── docs/
        │   ├── text
        │   └── python
        └── img
    ```
    
3. [x] Crea los archivos de la aplicación con el siguiente código (incluyelo en el script):
    
    ```sh
    for file in index.html style.css app.js main.py
    do
        touch ./"${file}"
    done
    ```

4. [x] Una vez creados los archivos la carpeta se vera asi utiliza el siguiente comando si quieres ver en consola tu estructura `cmd //c tree //F ./`:

    ```
    .
    └── webapp/
        ├── app.js
        ├── index.html
        ├── main.py
        ├── style.css
        ├── css
        ├── docs/
        │   ├── text
        │   └── python
        ├── img
        └── js

    ```

5. [x] Descargar la imagen de internet con el siguiente comando:
    
    ```sh
    curl -o meme.jpg "https://pm1.aminoapps.com/7021/417429318f6b29b6bc737fb95ba1ddafc9ba2decr1-720-642v2_hq.jpg"
    ```
6. [x] Guarda en una variable la cantidad de archivos de texto que creara el script y pon el siguiente código esto creara la cantidad de archivos .txt ingresada por consola:
    
    ```sh
    read -p "Introduce la cantidad de archivos a crear:" numfiles
    for x in $( seq 1 $numfiles )
    do
        echo "Archivo ${x}" > "file${x}.txt"
    done
    ```

7. [x] Verificar estructura con `"cmd //c tree //F ./"` debe verse de la siguiente manera

    ```
    .
    └── webapp/
        ├── app.js
        ├── file1.txt
        ├── file2.txt
        ├── file3.txt
        ├── file4.txt
        ├── file5.txt
        ├── ...
        ├── ...
        ├── fileN.txt
        ├── index.html
        ├── main.py
        ├── meme.jpg
        ├── style.css  
        ├── css
        ├── docs/
        │   ├── text
        │   └── python
        ├── img
        └── js

    ```
   
8. [x] En este punto crea un segundo commit con el mensaje *`"Second commit, structure created"`* que contiene la actualización del script (este comando no debe ser parte del script).
   
9. [x] En el script debe organizar la informacion moviendo los archivos a su respectiva carpeta usando el comando *`"mv"`*:
   
      - [x] Mueve el archivo *`style.css`* a la carpeta **`css`**
      - [x] Los archivos *`fileN.txt`* a la carpeta **`docs/text`**
      - [x] El archivo *`main.py`* a la carpeta **`docs/python`**
      - [x] El *`app.js`* a la carpeta **`js`** 
      - [x] Dejar *`index.html`* en la ruta actual, dentro de **`webapp`**. 
   
10. [x] Al ejecutar `"cmd //c tree //F ./"` debe verse asi y concluye la programación del script:
   
    ```
    .
    └── webapp/
        ├── index.html
        ├── css/
        │   └── style.css
        ├── docs/
        │   ├── text/
        │   │   ├── file1.txt
        │   │   ├── file2.txt
        │   │   ├── file3.txt
        │   │   ├── file4.txt
        │   │   ├── file5.txt
        │   │   ├── ...
        │   │   ├── ...
        │   │   └── fileN.txt
        │   └── python/
        │       └── main.py
        ├── img/
        │   └── meme.jpg
        └── js/
            └── app.js
        
    ```
---
- [x] Una vez completado el script ejecutarlo para validar que haga lo antes pedido: `./create.sh`
- [ ] Mover este archivo `examen372.md` a la carpeta **`examen372`**
- [ ] Renombrar el archivo `examen372.md` a README.md con la terminal *`"mv"`*
- [ ] Registrar un tercer commit con el mensaje *`"Third commit, add README.md and final script updated"`*
- [ ] Guardar el historial de sus comandos: *`history > history.txt`*
- [ ] Registrar el último commit agregando el archivo del historial con el mensaje *`"Fourth commit, command history added"`*
- [ ] Crear un repositorio remoto en sus cuentas de GITHUB con el nombre *`examen372`*
- [ ] Subir su carpeta al repositorio remoto con su famosisimo: *`git push`* 🔥👀

---

### Extras:

<span style="font-size:1.5em">**+1 extra**</span> en la calificación del examen si colocas el código siguiente en los archivos de la app.

> index.html

```html
<!DOCTYPE html>
<html lang="es">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Cheems I love you 💖🐶</title>
    <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Lato:wght@400;700&display=swap">
    <link rel="stylesheet" href="./css/styles.css">
    
</head>
<body>
    <main id="content">
        <img id="cheems-img" src="https://img.asmedia.epimg.net/resizer/zCBy1qXw2E26HUAAQTBOYB0eSko=/736x414/cloudfront-eu-central-1.images.arcpublishing.com/diarioas/KKF52TVGVNCKJFW6RGQRBR6OBE.png" alt="Perrito Cheems">
    </main>
    
    <script src="https://cdn.jsdelivr.net/npm/tsparticles-confetti@2.12.0/tsparticles.confetti.bundle.min.js"></script>
    <script src="./js/app.js"></script>
</body>
</html>
```

> style.css

```css
body {
    font-family: 'Roboto', sans-serif;
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    margin: 0;
    background-color: #333;
}

#cheems-img {
    cursor: pointer;
}

h1 {
    font-size: 4rem; /* Tamaño de fuente de 4rem */
    color: #fff; /* Texto blanco */
    text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.5); /* Sombra de texto */
}
```

> app.js
> 
```js
// Add envent click to image
const content = document.getElementById("content");
const cheemsImg = document.getElementById("cheems-img");
const h1 = document.createElement("h1");
h1.textContent = "¡Felicidades Cheems 😭!";

cheemsImg.addEventListener("click", function() {
    cheemsImg.style.display = "none"; // Hide image
    content.appendChild(h1); // Add h1 element
    explode();
    
});

// Function to make explode confetti
function explode() {
    const duration = 15 * 1000,
    animationEnd = Date.now() + duration,
    defaults = { startVelocity: 30, spread: 360, ticks: 60, zIndex: 0 };

    function randomInRange(min, max) {
    return Math.random() * (max - min) + min;
    }

    const interval = setInterval(function() {
    const timeLeft = animationEnd - Date.now();

    if (timeLeft <= 0) {
        return clearInterval(interval);
    }

    const particleCount = 50 * (timeLeft / duration);

    // since particles fall down, start a bit higher than random
    confetti(
        Object.assign({}, defaults, {
        particleCount,
        origin: { x: randomInRange(0.1, 0.3), y: Math.random() - 0.2 },
        })
    );
    confetti(
        Object.assign({}, defaults, {
        particleCount,
        origin: { x: randomInRange(0.7, 0.9), y: Math.random() - 0.2 },
        })
    );
    }, 250);
}
```

`Nota:` Si descargo tu script y ejecuta la app tienes <span style="font-size:1.5em">**+2 extra**</span>.
<br />
<br />
<br />
<div align="center" width="250" height="250">

![Examen](https://www.elorientadero.com/wp-content/uploads/2022/01/meme-1.jpg)

</div>

history.txt
  134  mkdir -p ./textoPlano
  135  mkdir -p ./pdf
  136  cd ./textoPlano
  137  touch textoPlano1.txt
  138  touch textoPlano2.txt
  139  cd ..
  140  cd./pdf
  141  cd ./pdf
  142  curl -o explicacion.pdf https://www.um.es/innova/OCW/informatica-para-universitarios/ipu_docs/la_shell/bash.pdf
  143  cd ..
  144  cd ..
  145  curl -o imagen1.jpg https://concepto.de/wp-content/uploads/2022/05/meme-e1653582504105.jpg
  146  mv imagen1.jpg ./MoverImagen1Aqui
  147  tree ./ > estructura.txt
  148  echo "el profe rifa y lo quiero mucho" >> estructura.txt
  149  tree ./
  150  history | tail -30 >> historial.txt
  151  cat historial.txt
  152  cd users
  153  cd c:
  154  cd users
  155  cd chuyh
  156  mkdir ciclos
  157  cd ciclos
  158  touch loops.sh
  159  chmod +x loops.sh
  160  nano loops.sh
  161  nano loops.sh
  162  nano loops.sh
  163  ./loops.sh
  164  nano loops.sh
  165  ./loops.sh
  166  nano loops.sh
  167  code loops.sh 
  168  git init
  169  git status
  170  git add .
  171  git commit -m "iniciamos proyecto y agregamos practica 1"
  172  git config --global user.jesus.hernandez.ruiz@uabc.edu.mx
  173  git config --global user.nam
  174  git config --global user.mail jesus.hernandez.ruiz@uabc.edu.mx
  175  git config --global user.name chuyH23
  176  git remote add origin https://github.com/chuyH23/PracticasDSW
  177  git push -u origin main
  178  status
  179  git status
  180  git status
  181  git status
  182  git status
  183  git push
  184  git add .
  185  git push
  186  git commit -m"cambio"
  187  git push
  188  cd users
  189  cd user
  190  cd c:
  191  cd users
  192  cd chuyh
  193  mkdir project1
  194  cd project1
  195  touch historia.txt
  196  code .
  197  git add historia.txt
  198  cd c:
  199  cd users
  200  cd chuyh
  201  git status
  202  git add .
  203  git comit -m "probamos etiquetas"
  204  git status
  205  git add .
  206  git status
  207  git add .
  208  git comit "aqui se prueban algunas etiquetas basicas"
  209  git comit -m"aqui se prueban algunas etiquetas basicas"
  210  git comit -m "aqui se prueban algunas etiquetas basicas"
  211  git commit -m "aqui se prueban algunas etiquetas basicas"
  212  git push
  213  git status
  214  git add .
  215  git commit -m "aqui se prueban tamaños y colores con html solamente"
  216  git push
  217  git status
  218  git add .
  219  git commit -m "aqui se prueba la funcion href"
  220  git push
  221  git status
  222  git add .
  223  git commit -m "aqui se prueba los links con una pagina a eleccion"
  224  git push
  225  git status
  226  git add .
  227  git commit -m "aqui se jugo con colores en html"
  228  git push
  229  git status
  230  git add .
  231  git commit -m "aqui se crea nuestra primera tabla"
  232  git push
  233  git status
  234  git add .
  235  git commit -m "aqui se creamos nuestro horario en una tabla"
  236  git push
  237  git status
  238  git add .
  239  git commit -m "aqui se creamos nuestro primer formulario"
  240  git push
  241  git status
  242  git add .
  243  git commit -m "aqui se creamos un formulario con ms variantes de etiquetas"
  244  git push
  245  git status
  246  git add .
  247  git commit -m "aqui se creamos una tabla con cellspading e imagenes"
  248  git push
  249  git status
  250  git add .
  251  git commit -m "aqui se creamos una tabla de hoteles combinando celdas"
  252  git push
  253  git status
  254  git add .
  255  git commit -m "aqui se creamos un formulario en una tabla"
  256  git push
  257  git status
  258  git add .
  259  git commit -m "aqui creamos un formulario en tabla donde se prueban etiquetas especiales para formulario"
  260  git push
  261  git status
  262  git add .
  263  git commit -m "aqui usamos css y probamos colores y secciones de las estructura html"
  264  git push
  265  git status
  266  git add .
  267  git commit -m "aqui probamos los id en los divs"
  268  git push
  269  git status
  270  git add .
  271  git commit -m "aqui se combinan y se ponen encabezados dentro de otros"
  272  git push
  273  git status
  274  git add .
  275  git commit -m "aqui se prueban las unidades de medidas que se pueden usar"
  276  git push
  277  git status
  278  git add .
  279  git commit -m "aqui se crea una pagina con los conocimientos que tengamos"
  280  git push
  281  git status
  282  git add .
  283  git commit -m "aqui seguimos las reglas segun un pdf que nos indivacaba acciones a seguir"
  284  git push
  285  git status
  286  git add .
  287  git commit -m "aqui probamos las 3 formas de tener una pagina"
  288  git push
  289  git status
  290  git add .
  291  git commit -m "iamgenes para las practicas"
  292  git push
  293  git status
  294  git add .
  295  git commit -m "primera practica con js donde probamos operaciones matematicas"
  296  git push
  297  git status
  298  git add .
  299  git commit -m "pruebas con condicionales y arrays en js"
  300  git push
  301  cd c:
  302  cd users
  303  cd chuyh
  304  mkdir scriptbash
  305  cd scriptbash
  306  code .
  307  ./ createp.sh
  308  ./createp.sh
  309  ./createp.sh
  310  ./createp.sh
  311  ./createp.sh
  312  ./createp.sh
  313  cd pythonClaseareass
  314  cd pythonClase
  315  cd ..
  316  cd pythonClase
  317  git init
  318  ls
  319  cd pythonClaseareass
  320  cd aaa
  321  ls
  322  status
  323  cd ..
  324  cd ..
  325  ./createp.sh
  326  cd scriptbash
  327  ./createp.sh
  328  cd..
  329  cd..
  330  cd ..
  331  cd chuyhreadmee
  332  ./createp.sh
  333  ./ createp.sh
  334  ./createp.sh
  335  ./createp.sh
  336  cd scriptbash
  337  ./createp.sh
  338  git status
  339  git init
  340  git status
  341  git add .
  342  git status
  343  git add .
  344  git commit -m "programa par e impar"
  345  git push
  346  git remote add origin https://github.com/chuyH23/onceProgramas
  347  git push
  348  git branch -m main
  349  git remote add origin https://github.com/chuyH23/onceProgramas

create.sh
#!/bin/bash


 rm -rf webapp
    mkdir webapp
    cd webapp
    mkdir css
    mkdir js
    mkdir docs
    cd docs
    mkdir text
    mkdir python
    cd ..
   mkdir img




for file in index.html style.css app.js main.py
    do
        touch ./"${file}"
    done

    curl -o meme.jpg "https://pm1.aminoapps.com/7021/417429318f6b29b6bc737fb95ba1ddafc9ba2decr1-720-642v2_hq.jpg"


    read -p "Introduce la cantidad de archivos a crear:" numfiles
    for x in $( seq 1 $numfiles )
    do
        echo "Archivo ${x}" > "file${x}.txt"
    done

    
   
   mv style.css css
   
    mv main.py docs/python
   
    mv app.js js
  
 
    mv file1.txt docs/text
   mv file2.txt docs/text
    mv file3.txt docs/text
    mv file4.txt docs/text
    mv file5.txt docs/text
 
 mv meme.jpg img
 history > history.txt



